src/lib.rs

```diff
@@ -47,13 +47,21 @@ impl Asset {
                 origin_path: origin_path.to_string(),
             })
         } else {
-            Ok(Asset::LocalAsset(LocalAsset::new(origin_path, contents)))
+            Ok(Asset::LocalAsset(LocalAsset::new(origin_path, contents)?))
         }
     }
 
+    /// Returns a computer filename based on origin path for both types of asset
+    pub async fn filename(origin_path: &str) -> Result<String> {
+        #[cfg(feature = "remote")]
+        if is_remote(origin_path)? {
+            return Ok(RemoteAsset::load(origin_path).await?.filename);
+        }
+        Ok(LocalAsset::load(origin_path)?.filename)
+    }
```

Comment:
I get that this is making a uniform interface but I can't help but feel like this is kinda a perf footgun. Like, we're downloading/reading an entire file just to report its name here, right?

(Not a blocker)

Reply:
so i don't disagree, but the trick is that we can't know it's name (for sure, in all cases) without fetching because a url doesn't always contain a filename even if it points at a file (a route name and it's response body could be entirely unrelated). we could provide an alternate impl where users provide the filename so we don't have to fetch?

Reply:
eliminated this api :)